mock-tikv/src/store.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use kvproto::kvrpcpb::KvPair;
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+};
+
+#[derive(Debug, Clone)]
+pub struct KvStore {
+    data: Arc<RwLock<HashMap<Vec<u8>, Vec<u8>>>>,
+}
+
+impl Default for KvStore {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl KvStore {
+    pub fn new() -> KvStore {
+        KvStore {
+            data: Arc::new(RwLock::new(HashMap::new())),
+        }
+    }
+
+    pub fn raw_get(&self, key: &[u8]) -> Option<Vec<u8>> {
+        let data = self.data.read().unwrap();
+        data.get(key).map(|v| v.to_vec())
+    }
+
+    pub fn raw_batch_get(&self, keys: &[Vec<u8>]) -> Vec<KvPair> {
+        let data = self.data.read().unwrap();
+        keys.iter()
+            .filter_map(|key| {
+                if data.contains_key(key) {
+                    let mut pair = KvPair::default();
+                    pair.set_value(data.get(key).map(|v| v.to_vec()).unwrap_or_else(Vec::new));
+                    pair.set_key(key.to_vec());
+                    Some(pair)
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
+
+    pub fn raw_put(&self, key: Vec<u8>, value: Vec<u8>) {
+        let mut data = self.data.write().unwrap();
+        data.insert(key, value);
+    }
+
+    pub fn raw_batch_put(&self, pairs: &[KvPair]) {
+        let mut data = self.data.write().unwrap();
+        for pair in pairs {
```

Comment:
Could use `data.extend(pairs.iter().map(...))`. Could also take `pairs` by value to avoid the clone.